test/test_router.go

```diff
@@ -26,13 +26,14 @@ func (s *RouterSuite) TestAdditionalHttpPorts(t *c.C) {
 	defer x.Destroy()
 
 	// Test that setting added HTTP and HTTPS ports succeeds
-	t.Assert(x.flynn("/", "-a", "router", "env", "set", "ADDITIONAL_HTTP_PORTS=8080"), Succeeds)
-	t.Assert(x.flynn("/", "-a", "router", "env", "set", "ADDITIONAL_HTTPS_PORTS=8081"), Succeeds)
+	watcher, err := x.controller.WatchJobEvents("router", "")
+	t.Assert(err, c.IsNil)
+	t.Assert(x.flynn("/", "-a", "router", "env", "set", "ADDITIONAL_HTTP_PORTS=8080", "ADDITIONAL_HTTPS_PORTS=8081"), Succeeds)
+	t.Assert(watcher.WaitFor(ct.JobEvents{"app": {ct.JobStateUp: 1, ct.JobStateDown: 1}}, 10*time.Second, nil), c.IsNil)
```

Comment:
No, for two reasons:

1. It's a `all-at-once` deploy, which means that the old formation is just scaled to zero at the end without waiting.
2. Even if it was a `one-by-one` deploy, the deployer waits for the discoverd registration to be pulled, not for the job to exit, so requests can still end up at the wrong router instance.